src/Models/Interfaces/Api.ts

```diff
@@ -3,5 +3,24 @@
 
 export enum ExtensionName {
   Toolkit = "Azure IoT Hub Toolkit",
-  AzureAccount = "Azure Account"
+  AzureAccount = "Azure Account",
+  Remote = "Remote Development",
+  AzureFunctions = "Azure Functions",
+  Arduino = "Arduino"
 }
+
+export class DependentExtensions {
+  static readonly azureFunctions = "ms-azuretools.vscode-azurefunctions";
+  static readonly arduino = "vsciot-vscode.vscode-arduino";
+  static readonly remote = "ms-vscode-remote.vscode-remote-extensionpack";
+  static readonly toolkit = "vsciot-vscode.azure-iot-toolkit";
+  static readonly azureAccount = "ms-vscode.azure-account";
+}
+
+export const ExtensionNameIdMap: Record<ExtensionName, DependentExtensions> = {

```

Comment:
why prefer Record<> to Map?

Reply:
Map and Record both work here.
For the difference, Map<K, V> does not require all members of type K to appear in the map, while Record<K, V> requires all members of K to appear in the record.
I change record to map here, since we don't have a strong restriction to ask all members of ExtensionName to appear in the map.

> Record<K, V> can be used to map the properties of a type to another type.

[Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt)